### PR TITLE
feat: implement list models in REST server

### PR DESCRIPTION
**Pull request by jeffmaury**
### What does this PR do?

Add list models operation to the REST server

Let's make @cescoffier happy !

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

#1296 

### How to test this PR?

Two ways:

- curl localhost:10434/api/tags
- OLLAMA_HOST=127.0.0.1:10434 ollama ls

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -52,12 +52,12 @@
     }
   },
   "scripts": {
-    "build": "vite build",
+    "build": "yarn openapi-typescript ../../api/openapi.yaml -o src-generated/openapi.ts && vite build",
```
Comment (benoitf) [inline packages/backend/package.json:55]: maybe use a "generate-open-ai" script and we call it from build and watch ?

File: packages/backend/src/managers/apiServer.ts [hunk 2]
```diff
@@ -26,14 +26,33 @@ import { existsSync } from 'fs';
 import { getFreeRandomPort } from '../utils/ports';
 import * as podmanDesktopApi from '@podman-desktop/api';
 import { readFile } from 'fs/promises';
+import type { ModelsManager } from './modelsManager';
+import type { components } from '../../src-generated/openapi';
```
Comment (benoitf) [inline packages/backend/src/managers/apiServer.ts:30]: looks like there are failure on the test

File: packages/backend/package.json (hunk 1 above)
Comment (jeffmaury) [reply packages/backend/package.json]: Fixed by following commit

File: packages/backend/src/managers/apiServer.ts (hunk 2 above)
Comment (jeffmaury) [reply packages/backend/src/managers/apiServer.ts:30]: Fixed

**Review by benoitf — APPROVED** [review]

**Comment by cescoffier** [comment]
\me happy :-) Thanks @jeffmaury

**Review by slemeur — APPROVED** [review]
Well done! Wonderful !!

**Pull request merged**